From f11186877a3d077cdf58992190c79fd9d77ac203 Mon Sep 17 00:00:00 2001
From: xtex <xtexchooser@duck.com>
Date: Tue, 6 Aug 2024 15:11:51 +0800
Subject: [PATCH] build: Find boost with name "Boost"

On many distributions like AOSC OS, Alpine Linux, Arch Linux, etc., boost should be searched with name "Boost".
This should fix the packaging issue on AOSC OS.

Signed-off-by: xtex <xtexchooser@duck.com>
---
 cmake/build_helpers.cmake | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/cmake/build_helpers.cmake b/cmake/build_helpers.cmake
index dce5eaae1b1d..d3bd92cd6a02 100644
--- a/cmake/build_helpers.cmake
+++ b/cmake/build_helpers.cmake
@@ -702,8 +702,13 @@ macro(addBundledLibraries)
     endif()
 
     if (USE_SYSTEM_BOOST)
-        find_package(boost REQUIRED)
-        set(BOOST_LIBRARIES boost::regex)
+        find_package(Boost CONFIG QUIET COMPONENTS regex)
+        if (Boost_FOUND)
+            set(BOOST_LIBRARIES Boost::regex)
+        else()
+            find_package(boost REQUIRED)
+            set(BOOST_LIBRARIES boost::regex)
+        endif()
     else()
         add_subdirectory(${THIRD_PARTY_LIBS_FOLDER}/boost ${CMAKE_CURRENT_BINARY_DIR}/boost EXCLUDE_FROM_ALL)
         set(BOOST_LIBRARIES boost::regex)

base-commit: 811214ddb79b66dba5ca56021212f81f9d89d608
-- 
2.46.0

